File: dashboard/src/types/scan.ts
export interface Vulnerability {
    id: string;
    severity: 'UNKNOWN' | 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';           // 'UNKNOWN' | 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
    cvss_score: number | null;
    description: string | null;
    link: string;
}
export interface PackageResult {
    name: string;
    current_version?: string;
    latest_version?: string;
    days_since_update: number | null;
    is_outdated: boolean;
    vulnerabilities: Vulnerability[];
    risk_level: 'GREEN' | 'YELLOW' | 'RED';           // 'GREEN' | 'YELLOW' | 'RED'
    action: string;
}
export interface ScanResult {
    packages: PackageResult[];
    audit_score: number;        // 0-100
    audit_level: 'GREEN' | 'YELLOW' | 'RED';          // 'GREEN' | 'YELLOW' | 'RED'
    breakdown: Record<string, number>;           
}
export interface TrafficResult {
    packages: PackageResult[];
    audit_score: number;        // 0-100
    audit_level: 'GREEN' | 'YELLOW' | 'RED';          // 'GREEN' | 'YELLOW' | 'RED'
    breakdown: Record<string, number>;           
}
export interface TextScanRequest {
    content: string;
}

// network analysis

export interface AnomalyEvent {
    timestamp: string;
    type: 'HIGH_LATENCY' | 'HIGH_BYTES';
    value: number;
    threshold: number;
}
export interface TimePoint {
    timestamp: string;
    avg_latency: number;
    max_latency: number;
    total_bytes: number;
    max_bytes: number;
    event_count: number;
    latency_anomaly_count: number;
    bytes_anomaly_count: number;
}
export interface EndpointSummary {
    ip: string;
    protocol: string;
    total_bytes: number;
    request_count: number;
    is_suspicious: number;
}
export interface Baseline {
    latency_mean: number;
    latency_std: number;
    bytes_mean: number;
    bytes_std: number;
    events_per_min: number;
    latency_threshold: number;
    bytes_threshold: number;
}
export interface TrafficAnalysisResult{
    baseline: Baseline;
    anomalies: AnomalyEvent[];
    timeseries: TimePoint[];
    endpoints: EndpointSummary[];
    network_risk_score: number;      // el N del Audit-Score
    total_events: number;
}

File: api/models/scan.py
from pydantic import BaseModel
from typing import Optional

class Vulnerability(BaseModel):
    id: str
    severity: str           # 'UNKNOWN' | 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
    cvss_score: float|None
    description: str|None
    link: str

class PackageResult(BaseModel):
    name: str
    current_version: Optional[str]
    latest_version: Optional[str]
    days_since_update: int|None
    is_outdated: bool
    vulnerabilities: list[Vulnerability]
    risk_level: str           # 'GREEN' | 'YELLOW' | 'RED'
    action: str

class ScanResult(BaseModel):
    packages: list[PackageResult]
    audit_score: int          # 0-100
    audit_level: str          # 'GREEN' | 'YELLOW' | 'RED'
    breakdown: dict           

class TextScanRequest(BaseModel):
    content: str

#network classes

class AnomalyEvent(BaseModel):
    timestamp: str
    type: str        # 'HIGH_LATENCY' | 'HIGH_BYTES'
    value: float
    threshold: float

class TimePoint(BaseModel):
    timestamp: str
    avg_latency: float
    max_latency: float
    total_bytes: int
    max_bytes: int
    event_count: int
    latency_anomaly_count: int
    bytes_anomaly_count: int

class EndpointSummary(BaseModel):
    ip: str
    protocol: str
    total_bytes: int
    request_count: int
    is_suspicious: bool

class Baseline(BaseModel):
    latency_mean: float
    latency_std: float
    bytes_mean: float
    bytes_std: float
    events_per_min: float
    latency_threshold: float
    bytes_threshold: float

class TrafficAnalysisResult(BaseModel):
    baseline: Baseline
    anomalies: list[AnomalyEvent]
    timeseries: list[TimePoint]
    endpoints: list[EndpointSummary]
    network_risk_score: float      # el N del Audit-Score
    total_events: int